pheno_data/03-dx_extract_dataset_R.qmd

```diff
@@ -0,0 +1,295 @@
+---
+jupyter: ir
+---
+
+# “dx extract_dataset” in R
+<hr/>
+***As-Is Software Disclaimer***
+
+This content in this repository is delivered “As-Is”. Notwithstanding anything to the contrary, DNAnexus will have no warranty, support, liability or other obligations with respect to Materials provided hereunder.
+
+<hr/>
+
+This notebook demonstrates usage of the dx command `extract_dataset` for:
+* Retrieval of Apollo-stored data, as referenced within entities and fields of a Dataset or Cohort object on the platform
+* Retrieval of the underlying data dictionary files used to generate a Dataset object on the platform
+
+<a href="https://github.com/dnanexus/OpenBio/blob/master/LICENSE.md">MIT License</a> applies to this notebook.
+
+## Preparing your environment
+### Launch spec:
+
+* App name: JupyterLab with Python, R, Stata, ML ()
+* Kernel: R
+* Instance type: Spark Cluster - mem1_ssd1_v2_x2, 3 nodes 
+* Snapshot: `/.Notebook_snapshots/jupyter_snapshot.gz`
+* Cost: < $0.2
+* Runtime: =~ 10 min
+* Data description: Input for this notebook is a v3.0 Dataset or Cohort object ID
+
+### Install dxpy
+extract_dataset requires dxpy version >= 0.329.0. If running the command from your local environment (i.e. off of the DNAnexus platform), it may be required to also install pandas. For example, pip3 install -U dxpy[pandas]
+
+```{r}
+#| trusted: true
+system("pip3 show dxpy", intern = TRUE)
+```
+
+### Install tidyverse for data processing
+
+Quick note - you will need to read the licenses for the tidyverse in order to make sure whether you and your group are comfortable with the licensing terms.
+
+```{r}
+#| trusted: true
+#| eval: false
+install.packages(c("readr", "stringr", "dplyr", "glue", "reactable", "janitor", "remotes"))
+remotes::install_github("laderast/xvhelper")
+```
+
+### Import packages
+
+```{r}
+#| trusted: true
+library(dplyr)
+library(readr)
+library(stringr)
+library(glue)
+library(reactable)
+library(xvhelper)
+```
+
+### 1. Assign environment variables
+
+```{r}
+#| trusted: true
+# The referenced Dataset is private and provided only to demonstrate an example input. The user will need to supply a permissible and valid record-id
+# Assign project-id of dataset
+# Assign dataset record-id
+rid <- 'record-G406j8j0x8kzxv3G08k64gVV'
+# Assign joint dataset project-id:record-id
+dataset <- rid
```

Comment:
Looks like you're missing the "project-id" here. Perhaps this is intentional. Maybe specify that the user needs to fill this information and where they can find this information. If that is the case, it might be better to have a section called "User inputs" or something similar

Reply:
Ok, will fix